src/config.py

```diff
@@ -1,89 +1,130 @@
-import toml
 import os
 
+import toml
+from fastlogging import LogInit
+from toml import TomlDecodeError
+
 
 class Config:
     _instance = None
+    _logger = None
 
     def __new__(cls):
-        if cls._instance is None:
-            cls._instance = super().__new__(cls)
-            cls._instance._load_config()
-        return cls._instance
-
-    def _load_config(self):
-        # If the config file doesn't exist, copy from the sample
-        if not os.path.exists("config.toml"):
-            with open("sample.config.toml", "r") as f_in, open("config.toml", "w") as f_out:
-                f_out.write(f_in.read())
-
-        self.config = toml.load("config.toml")
+        if cls._instance:
+            return cls._instance
+
+        cls._instance = super().__new__(cls)
+        cls._logger = LogInit(pathName="logs/core.log", console=True, colors=True)
```

Comment:
log path is fetch from config file

Reply:
We can't fetch from config file here as it needs to handle the case when the config file is missing

Reply:
sample.config

Reply:
I think it would be better of we can avoid IO here as there won't be any way to handle any errors as not even the logging will be enabled. Like what if the user incorrectly edits the sample.config and there is some mistake. 

If you are sure that this won't be a concern I can update the code.

Reply:
Or if the user renamed config.example to config etc

Reply:
also why creating new logger? just fetch from logger file and log it. so it will be in one file and simple